pkg/kubelet/server/server.go

```diff
@@ -209,11 +212,27 @@ func ListenAndServeKubeletReadOnlyServer(
 	}
 }
 
+type PodResourcesProviders struct {
+	Pods    podresources.PodsProvider
+	Devices podresources.DevicesProvider
+	Cpus    podresources.CPUsProvider
+	Memory  podresources.MemoryProvider
+}
+
 // ListenAndServePodResources initializes a gRPC server to serve the PodResources service
-func ListenAndServePodResources(socket string, podsProvider podresources.PodsProvider, devicesProvider podresources.DevicesProvider, cpusProvider podresources.CPUsProvider, memoryProvider podresources.MemoryProvider) {
-	server := grpc.NewServer()
-	podresourcesapiv1alpha1.RegisterPodResourcesListerServer(server, podresources.NewV1alpha1PodResourcesServer(podsProvider, devicesProvider))
-	podresourcesapi.RegisterPodResourcesListerServer(server, podresources.NewV1PodResourcesServer(podsProvider, devicesProvider, cpusProvider, memoryProvider))
+func ListenAndServePodResources(socket string, providers PodResourcesProviders, ratelimit kubeletconfiginternal.RateLimitConfiguration) {
+	maxCallsLimit := gotimerate.Inf
+	maxCallsBurst := 0
+	if ratelimit.MaxFrequency > 0 {
+		maxCallsLimit = gotimerate.Limit(ratelimit.MaxFrequency)
+		maxCallsBurst = int(ratelimit.MaxBurst)
```

Comment:
I think we should add a sanity check on MaxBurst (to ensure that MaxBurst is > 0) given my earlier comment - https://github.com/kubernetes/kubernetes/pull/115852/files#r1132523091

Reply:
Good point, will add.